x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/TransportUpdateConnectorFilteringAction.java

```diff
@@ -47,6 +52,23 @@ protected void doExecute(
         UpdateConnectorFilteringAction.Request request,
         ActionListener<ConnectorUpdateActionResponse> listener
     ) {
-        connectorIndexService.updateConnectorFiltering(request, listener.map(r -> new ConnectorUpdateActionResponse(r.getResult())));
+        String connectorId = request.getConnectorId();
+        List<ConnectorFiltering> filtering = request.getFiltering();
+        FilteringAdvancedSnippet advancedSnippet = request.getAdvancedSnippet();
+        List<FilteringRule> rules = request.getRules();
+        if (request.getFiltering() == null) {
```

Comment:
To me it's not immediately obvious why we update `draft`, if `filtering` is `null`? Maybe an explanatory comment would help to understand this behavior.